six/projects/csm/source/SIXSensorModel.cpp

```diff
@@ -1110,5 +1111,21 @@ double SIXSensorModel::getCorrelationCoefficient(size_t cpGroupIndex,
                          "SIXSensorModel::getCorrelationCoefficient");
     }
 }
+
+DataType SIXSensorModel::getDataType(const csm::Des& des)
+{
+    std::string desid = des.subHeader().substr(NITF_DE_SZ, NITF_DESTAG_SZ);
+    str::trim(desid);
+    const std::string& subheader = des.subHeader();
+
+    const size_t desshsiOffset = 73 + 200;
+    const size_t desshsiLength = 60;
+    std::string desshsiField = des.subHeader().substr(
+            desshsiOffset, desshsiLength);
+    str::trim(desshsiField);
+
+    return NITFReadControl::getDataType(desid, subheader.size() - 200,
```

Comment:
Well, since you already need to be dealing with this as a string anyhow... you can use the fourth argument and send in the DESID.  I think this should be bytes [2, 27] of the string.